rest/data/task.go

```diff
@@ -84,12 +85,7 @@ func FindTasksByProjectAndCommit(opts task.GetTasksByProjectAndCommitOptions) ([
 				break
 			}
 		}
-		if !found {
-			return []task.Task{}, gimlet.ErrorResponse{
-				StatusCode: http.StatusNotFound,
-				Message:    fmt.Sprintf("task '%s' not found", opts.StartingTaskId),
-			}
-		}
+		grip.DebugWhen(!found, fmt.Sprintf("starting task '%s' not found", opts.StartingTaskId))
```

Comment:
We should avoid logging debug statements we don't have a plan for -- without a ticket number and/or planned investigation for it, we put logging in splunk that we don't actually need. 

If you think this will be helpful for future debugging, then I would include the current ticket number, and then not close the current ticket until you feel confident this change fixed the problem, and then remove the logging. Otherwise, removing the whole for loop is more useful.

Reply:
Yeah, I'm pretty sure this is the part where the route returns 404 in the way that Brooke described so not returning an error should "solve" the issue. 
The problem is that I can't figure out why the result of the loop would suddenly change from not being able to find the task to the route working normally. 

I created a ticket to follow up 
https://jira.mongodb.org/browse/EVG-17897

Reply:
Yeah I'm not sure either but this loop does seem a bit unnecessary so seems worth retrying. 
But by "include the ticket number" I meant to format grip statements in a meaningful way using message.Fields.

I would also ask that you comment on the ticket to let Brooke know that you're trying something but it may not fix the issue and she should let you know if she sees it again, in which case you'll re-open the existing ticket (or don't close the existing ticket for a few days).

Reply:
Sorry, I misunderstood. 
I updated brook on the ticket and changed the debug message!